File: Ahorn/entities/EntityWrappers/EntityMuter.jl
module VivHelperEntityMuter
using ..Ahorn, Maple

@mapdef Entity "VivHelper/EntityMuter" EntityMuter(x::Integer, y::Integer, width::Integer=8, height::Integer=8, Types::String="", all::Bool=true)

const placements = Ahorn.PlacementDict(
    "Entity Muter (Viv's Helper)" => Ahorn.EntityPlacement(
        EntityMuter,
        "rectangle"
    )
)

Ahorn.resizable(entity::EntityMuter) = true, true
Ahorn.minimumSize(entity::EntityMuter) = 8,8

Ahorn.selection(entity::EntityMuter) = Ahorn.getEntityRectangle(entity)

sprite = "ahorn/VivHelper/mute.png"

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::EntityMuter, room::Maple.Room)
    Ahorn.drawRectangle(ctx, 0, 0, entity.data["width"], entity.data["height"], (0.4,0.4,0.4,0.4))
    Ahorn.drawSprite(ctx, sprite, entity.data["width"]/2, entity.data["height"]/2)
end

end

File: Loenn/entities/EntityWrappers/EntityMuter.lua
local drawableRect = require('structs.drawable_rectangle')
local drawableSprite = require('structs.drawable_sprite')

return {
    name = "VivHelper/EntityMuter",
    placements = {
        name = "main",
        data = {
            width = 8, height = 8, 
            Types = "", all = true
        }
    },
    sprite = function(room,entity) return {
        drawableRect.fromRectangle("fill",entity.x,entity.y,entity.width,entity.height,{0.4,0.4,0.4,0.4}),
        drawableSprite.fromTexture('ahorn/VivHelper/mute', entity):setPosition(entity.x + entity.width/2, entity.y+ entity.height/2)
    } end
    
}